### PR TITLE
Bugfix NDC CO2 emission pricing in coupled and standalone runs

### PR DESCRIPTION
**Pull request by flohump**
## :bird: Description of this PR :bird:

Summary: 
- New default for AFOLU GHG emission pricing is no GHG emission pricing in the AFOLU sector before (and including) 2030. This setting will be also used in coupled REMIND-MAgPIE runs.
- For guiding land-use decisions in case of NDC afforestation policy (strong increase in 2025), we assume a small CO2 price of 5 USD per ton CO2 in 2025 and 2030 in case NDC afforestation is active.

### changed
- **56_ghg_policy** renamed `cfg$mute_ghgprices_until` to `cfg$gms$c56_mute_ghgprices_until` and changed the default to `y2030`, i.e. no GHG emission pricing in the AFOLU sector before (and including) 2030. This setting will be also used in coupled REMIND-MAgPIE runs.

### added
- **32_forestry** added interface `sm_ndc_aff_policy`

### removed
- **56_ghg_policy** removed `s56_ghgprice_phase_in` and `s56_ghgprice_start`
- **scripts** removed argument `mute_ghgprices_until`, now handeld in GAMS code

### fixed
- **56_ghg_policy** the switch `c56_mute_ghgprices_until` is now always used for coupled as well as standalone runs. 
- **56_ghg_policy** small CO2 price of 5 USD per tCO2 (18 USD per tC) in 2025 and 2030 in case of NDC afforestation `sm_ndc_aff_policy` to guide land-use decisions

## :wrench: Checklist for PR creator :wrench:

- [x] Label pull request [from the label list](https://github.com/magpiemodel/magpie/labels).
  - **Low risk**: Simple bugfixes (missing files, updated documentation, typos) or changes  in start or output scripts
  - **Medium risk**: Uncritical changes in the model core (e.g. moderate modifications in non-default realizations)
  - **High risk**: Critical changes in model core or default settings (e.g. changing a model default or adjusting a core mechanic in the model)

Emissions_CO2_Land_Land_use_Change-66
Emissions_CO2_Land_Cumulative_Land_use_Change-9
Prices_GHG_Emission_CO2_Primary_Forest_Vegetation_carbon-5
SDG_SDG15_Afforestation-2

- [x] Self-review own code
  - No hard coded numbers and cluster/country/region names.
  - The new code doesn't contain declared but unused parameters or variables.
  - [`magpie4`](https://github.com/pik-piam/magpie4) R library has been updated accordingly and backwards compatible where necessary.
  - `scenario_config.csv` has been updated accordingly (important if `default.cfg` has been updated)

- [x] Document changes 
  - Add changes to `CHANGELOG.md`
  - Where relevant, put In-code documentation comments
  - Properly address updates in interfaces in the module documentations
  - run [`goxygen::goxygen()`](https://github.com/pik-piam/goxygen) and verify the modified code is properly documented

- [x] Perform test runs
  - **Low risk**: 
    - Run a compilation check via `Rscript start.R --> "compilation check"`
  - **Medium risk**: 
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
  - **High risk**:
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
    - Default run from the PR target branch for comparison
    - Provide relevant comparison plots (land-use, emissions, food prices, land-use intensity,...)

### :chart_with_downwards_trend: Performance changes :chart_with_upwards_trend:
  
  - Current develop branch default : unchanged
  - This PR's default :  unchanged

## :rotating_light: Checklist for reviewer :rotating_light:

- PR is labeled correctly
- Code changes look reasonable
  - No hard coded numbers and cluster/country/region names.
  - No unnecessary increase in module interfaces
  - model behavior/performance is satisfactory.
- Changes are properly documented
  - `CHANGELOG` is updated correctly
  - Updates in interfaces have been properly addressed in the module documentations
  - In-code documentation looks appropriate
- [x] content review done (at least 1)
- [ ] RSE review done (at least 1)

### CODE REVIEW

**Review by mishkos — COMMENTED** [review]
Looks good to me now. One question remains, if we give a small co2 price to assure that NDC afforestation is correctly implemented (i.e. not on other land), what is the reason we don't have this for NPI afforestation too?

**Comment by flohump** [comment]
> Looks good to me now. One question remains, if we give a small co2 price to assure that NDC afforestation is correctly implemented (i.e. not on other land), what is the reason we don't have this for NPI afforestation too?

I consider the small CO2 price as part of the much more ambitious NDC package.

**Review by mishkos — APPROVED** [review]
Ok, that's plausible. We wouldn't maybe see past difforestation if we had small carbon tax in NPIs. 

All good!

File: modules/32_forestry/dynamic_feb21/input.gms [hunk 1]
```diff
@@ -47,6 +47,7 @@ scalars
   s32_aff_bii_coeff               BII coefficent to be used for CO2 price driven afforestation 0=natural vegetation 1=plantation (1) / 0 /
   s32_max_aff_area_glo            Switch for global or regional afforestation constraint (1) / 1 /
   s32_aff_prot                    Switch for protection of afforested areas (0=until end of planning horizon 1=forever) / 0 /
+  sm_ndc_aff_policy               Interface for NDC afforestation policy / 0 /
```
Comment (dklein-pik) [inline modules/32_forestry/dynamic_feb21/input.gms:50]: This switch is only used in module 32, right? If so could you renamed to `s32_ndc_aff_policy` please?

Comment (flohump) [reply modules/32_forestry/dynamic_feb21/input.gms:50]: No, this interface is used in 56_ghg_policy.

File: modules/32_forestry/dynamic_feb21/preloop.gms [hunk 2]
```diff
@@ -219,6 +219,12 @@ loop(j,
 *p32_land_start_ac(j,type32,ac) = p32_land("y1995",j,type32,ac);
 
 *** NPI/NDC policies BEGIN
+$ifthen "%c32_aff_policy%" == "ndc"
+ sm_ndc_aff_policy = 1;
```
Comment (tscheypidi) [inline modules/32_forestry/dynamic_feb21/preloop.gms:223]: these switches should be set via the config and the code should not configure itself. Why is that needed here?

File: modules/32_forestry/dynamic_feb21/input.gms (hunk 1 above)
Comment (dklein-pik) [reply modules/32_forestry/dynamic_feb21/input.gms:50]: Oh, sorry, missed that.

File: modules/56_ghg_policy/price_aug22/preloop.gms [hunk 3]
```diff
@@ -29,14 +29,18 @@ $elseif "%c56_pollutant_prices%" == "emulator"
...
 
+** Harmonize based on c56_mute_ghgprices_until
+loop(t_all$(m_year(t_all) <= m_year("%c56_mute_ghgprices_until%")),
+im_pollutant_prices(t_all,i,pollutants,emis_source) = 0;
+);
 
+** low C price in case of NDC afforestation policy
+if(sm_ndc_aff_policy = 1,
+  im_pollutant_prices("y2025",i,"co2_c",emis_source) = s56_cprice_ndc;
+  im_pollutant_prices("y2030",i,"co2_c",emis_source) = s56_cprice_ndc;
```
Comment (tscheypidi) [inline modules/56_ghg_policy/price_aug22/preloop.gms:42]: Shouldn't that also somehow be linked to mute_ghgprices_until? Here the pricing is hardlinked to 2025 and 2030.

In addition, it is somehow contradictory. We first set the prices to 0 and then we reset them again to a value. 

Alternative suggestion: 

1. convert s56_cprice_ndc to s56_minimum_cprice and use it as level to set it to, when mute_ghgprices is being used
2. set this via the scenario_config in the NDC scenario to 18
3. remove the cross-dependency between aforestation and ghgpolicy module

**Review by tscheypidi — CHANGES_REQUESTED** [review]

File: modules/32_forestry/dynamic_feb21/input.gms (hunk 1 above)
Comment (flohump) [reply modules/32_forestry/dynamic_feb21/input.gms]: removed in the revised version

File: modules/32_forestry/dynamic_feb21/preloop.gms (hunk 2 above)
Comment (flohump) [reply modules/32_forestry/dynamic_feb21/preloop.gms]: Done. Now in the scenario config

File: modules/56_ghg_policy/price_aug22/preloop.gms (hunk 3 above)
Comment (flohump) [reply modules/56_ghg_policy/price_aug22/preloop.gms]: Revised. Now depending on sm_fix_ssp2 and mute_ghgprices_until

File: config/default.cfg [hunk 4]
```diff
@@ -1381,9 +1375,13 @@ cfg$gms$c56_pollutant_prices_noselect <- "R21M42-SSP2-NPi"     # def = R21M42-SS
...
 
-# * Set GHG prices from the exogenous file (see above) to zero until (and including)
-# * the year given here
-cfg$mute_ghgprices_until <- "y2010"
+# * Mute GHG prices until (and including) the year given here
+# * Note: minimum C price (see below) will be set regardless of this setting
+cfg$gms$c56_mute_ghgprices_until <- "y2030"
+
+# * Minium C price for future time steps (USD per tC) 
+# * Note: used to guide land-use decisions in case of NDC policy (18 USD per tC corespond to roughly 5 USD per tCO2)
```
Comment (tscheypidi) [inline config/default.cfg:1383]: please clarify that the minimum price is only applied for years in the future which lie before mute_ghgprices_until. Otherwise one would expect that also every else too low co2 prices would get corrected

File: config/default.cfg [hunk 5]
```diff
@@ -1381,9 +1375,13 @@ cfg$gms$c56_pollutant_prices_noselect <- "R21M42-SSP2-NPi"     # def = R21M42-SS
...
+cfg$gms$s56_minimum_cprice <- 0
```
Comment (tscheypidi) [inline config/default.cfg:1384]:
```suggestion
cfg$gms$s56_minimum_cprice <- 0    # def = 0
```

**Review by tscheypidi — APPROVED** [review]
looks good. Please only update the description of s56_minimum_cprice

File: config/default.cfg (hunk 4 above)
Comment (flohump) [reply config/default.cfg]: Done.
